src/lib/y2network/sysconfig/connection_config_writers/ethernet.rb

```diff
@@ -17,28 +17,23 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
+require "y2network/sysconfig/connection_config_writers/base"
+
 module Y2Network
   module Sysconfig
     module ConnectionConfigWriters
       # This class is responsible for writing the information from a ConnectionConfig::Ethernet
       # object to the underlying system.
-      class Ethernet
-        # @return [Y2Network::Sysconfig::InterfaceFile]
-        attr_reader :file
-
-        def initialize(file)
-          @file = file
-        end
-
+      class Ethernet < Base
         # Writes connection information to the interface configuration file
         #
         # @param conn [Y2Network::ConnectionConfig::Base] Configuration to write
         def write(conn)
           file.bootproto = conn.bootproto.name
-          file.ipaddr = conn.ip_address
           file.name = conn.description
           file.startmode = conn.startmode.to_s
           file.ifplugd_priority = conn.startmode.priority if conn.startmode.name == "ifplugd"
+          write_ip_configs(conn.ip_configs)
```

Comment:
why not simple super that do all shared writtings like that ip_configs and probably also startmore or name?